.github/workflows/source-and-docs-release.yml

```diff
@@ -0,0 +1,100 @@
+on:
+  workflow_dispatch:
+    inputs:
+      git_remote:
+        type: choice
+        description: "Git remote to checkout"
+        options:
+          - python
+          - Yhg1s
+          - pablogsal
+          - ambv
+      git_commit:
+        type: string
+        description: "Git commit to target for the release. Must use the full commit SHA, not the short ID"
+      cpython_release:
+        type: string
+        description: "CPython release number (ie '3.11.5', note without the 'v' prefix)"
+
+name: "Build Python source and docs artifacts"
+
+jobs:
+  source-and-docs:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: "Checkout python/release-tools"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: "Checkout ${{ inputs.git_remote }}/cpython"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          repository: "${{ inputs.git_remote }}/cpython"
+          ref: "v${{ inputs.cpython_release }}"
+          path: "cpython"
+
+      - name: "Verify CPython commit matches tag"
+        run: |
+          if [[ "${{ inputs.git_commit }}" != "$(cd cpython && git rev-parse HEAD)" ]]; then
+            echo "expected git commit ('${{ inputs.git_commit }}') didn't match tagged commit ('$(git rev-parse HEAD)')"
+            exit 1
+          fi
+
+      - name: "Setup Python"
+        uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236 # v4.7.1
+        with:
+          python-version: 3.11
```

Comment:
Probably should update this to `3.x`? Wish there was a way to track this with Dependabot.